crates/core/src/doc/mod.rs

```diff
@@ -29,4 +31,16 @@ mod lives; // Processes any live queries relevant for this document
 mod pluck; // Pulls the projected expressions from the document
 mod purge; // Deletes this document, and any edges or indexes
 mod store; // Writes the document content to the storage engine
-mod table; // Processes any foreign tables relevant for this document
+mod table; // Processes any foreign tables relevant for this document'
+
+/// Error result used when a function can result in the value being processed being ignored.
+pub enum IgnoreError {
```

Comment:
Shouldn't we use a different result?

In fact, Ignore is a result not an error?

```
pub enum DocumentResult {
   Ok,
    Ignore,
    Error(Box<err::Error>)
}
```

Reply:
This won't work with `?` and would need manual raising when ever it is passed up. If `Try` was stable I would have implemented it that way instead but it currently isn't and I considered it introducing too much boilerplate.

Reply:
The approach here is the same as the one I used for the control flow refactor. Whenever `Try` gets stabilized I would suggest we change it to be a single enum but for now I think this is the best approach.